### PR TITLE
Script for transformation edx usual problem into adaptive capa problem

### PR DESCRIPTION
**Pull request by idegtiarov**

### CODE REVIEW

File: tools/transform_problems.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python
+"""
```
Comment (z4y4ts) [inline tools/transform_problems.py]: This docstring is pretty long. Please extract it to separate README.md

File: tools/transform_to_adapt/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# Script to transform problem into adaptive capa problem.
+
+Currently script supports transformation for two main problems: `Multiple Choice` and `Checkbox`.
+For example Checkbox problem::
```
Comment (z4y4ts) [inline tools/transform_to_adapt/README.md]: :: -> :

File: tools/transform_to_adapt/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      <compoundhint value="A B">Feedback 5</compoundhint>
+      <compoundhint value="A C">Feedback 6</compoundhint>
+      <compoundhint value="B C">Feedback 8</compoundhint>
+      <compoundhint value="A B C">Feedback 11</compoundhint>
+    </checkboxgroup>
+  </choiceresponse>
+</problem>
+```
+
+Would be transform into adaptive capa problem::
```
Comment (z4y4ts) [inline tools/transform_to_adapt/README.md]: :: -> :

File: tools/transform_to_adapt/transform_problems.py [hunk 4]
```diff
@@ -302,13 +243,36 @@ def transform_problem():
...
+    output_file = os.path.join(output_dir, os.path.split(problem_file)[1])
     with open(output_file, 'wb') as fhandler:
         fhandler.write(pretty_print(adaptive_problem).encode('utf-8'))
 
 
+def processing_problem_source(params, dir_walking, capa_template):
+    """
+    Processing transformation for chosen file(s) from the problems source
+
+    :param params: script expected parameters
```
Comment (z4y4ts) [inline tools/transform_to_adapt/transform_problems.py]: What type is that supposed to be?

File: tools/transform_to_adapt/transform_problems.py [hunk 5]
```diff
@@ -302,13 +243,36 @@ def transform_problem():
...
+            transform_problem(os.path.join(root, problem), params.output_dir, capa_template)
+
+
 def main():
-    transform_problem()
+    parameters = get_parameters()
+    dir_walking, capa_template = check_problems_source(parameters)
+
+    processing_problem_source(parameters, dir_walking, capa_template)
+    print("Transformation done! Adaptive problems are in the {} dir.".format(parameters.output_dir))
```
Comment (z4y4ts) [inline tools/transform_to_adapt/transform_problems.py]: `Adaptive problems are in the {} dir` -> `Your can find course with converted problems in {} directory.`

File: tools/transform_to_adapt/transform_problems.py [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    text = ('{0:>{indent}}{prefix}' + ('&lt;{tag}{attrib}&gt;' if tagwrap else '') + '{text}').format(
+        "", indent=indent, prefix=prefix, tag=tag, attrib=attrib, text=text_param
+    )
+    if len(element) > 0:
+        for subelement in element:
+            if subelement.tag == restrict_elem:
+                continue
+            text += parse_text_field(subelement)
+            if subelement.tail:
+                print subelement.tail
```
Comment (z4y4ts) [inline tools/transform_to_adapt/transform_problems.py]: Use print function to be future-proof.

File: tools/transform_to_adapt/README.md (hunk 2 above)
Comment (idegtiarov) [inline tools/transform_to_adapt/README.md]: done

File: tools/transform_to_adapt/README.md (hunk 3 above)
Comment (idegtiarov) [inline tools/transform_to_adapt/README.md]: done

File: tools/transform_to_adapt/transform_problems.py (hunk 6 above)
Comment (idegtiarov) [inline tools/transform_to_adapt/transform_problems.py]: It is debug string, removed

File: tools/transform_to_adapt/transform_problems.py (hunk 4 above)
Comment (idegtiarov) [inline tools/transform_to_adapt/transform_problems.py]: add object type

**Pull request merged**